### PR TITLE
Add with-chatbase example

### PR DESCRIPTION
**Pull request by xxhomey19**

### CODE REVIEW

File: examples/with-chatbase/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+The following code is the main part of this example which demonstrates the way
+to connect your bot with [Chatbase](https://chatbase.com/welcome). Furthermore,
+You have better check
+[middleware guide](https://bottender.js.org/docs/Guides-Middleware) if you don't
+understand what is middleware.
+
+```js
+const server = createServer(bot, {
+  verifyToken: '__FILL_YOUR_VERIFY_TOKEN_HERE__',
+  webhookMiddleware: botimizeMiddleware(bot, {
```
Comment (kpman) [inline examples/with-chatbase/README.md]: chatbaseMiddleware

File: examples/with-chatbase/chatbaseMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const _chatbase = require('@google/chatbase');
+
+module.exports = function botimizeMiddleware(bot, { apiKey, platform }) {
```
Comment (kpman) [inline examples/with-chatbase/chatbaseMiddleware.js]: chatbaseMiddleware

File: examples/with-chatbase/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Idea of this example
+
+This example shows how to combine your bot with **bot analytics tool**. In this
+case, we take [Chatbase](https://chatbase.com/welcome) which is built by Google
+as an example.
+
+The following code is the main part of this example which demonstrates the way
+to connect your bot with [Chatbase](https://chatbase.com/welcome). Furthermore,
+You have better check
+[middleware guide](https://bottender.js.org/docs/Guides-Middleware) if you don't
```
Comment (tw0517tw) [inline examples/with-chatbase/README.md]: `You have better ...` sounds not so polite.
This sentence can be rewrite as
> You can check middleware guide for more information

Comment (chentsulin) [reply examples/with-chatbase/README.md]: We should talk about express middleware here, not our middleware guide.

**Pull request merged**